crates/polars-pipe/src/executors/sinks/group_by/aggregates/mean.rs

```diff
@@ -121,6 +131,14 @@ where
         if let Some(val) = self.sum {
             unsafe {
                 match K::PRIMITIVE {
+                    PrimitiveType::Int64 => {
+                        let mut arr = val.to_i64().unwrap_unchecked_release();
+                        if self.as_date {
+                            let ms_in_day = 86_400_000i64;
+                            arr *= ms_in_day;
+                        }
+                        AnyValue::Int64(arr / self.count as i64)
+                    },
```

Comment:
I'm not familiar with this part of the codebase - do you fancy explaining why this is necessary?

What currently happens for `Datetime`, is that already taken care of somewhere else? (because it wouldn't match the float32 / float64 arms below, right?)

Reply:
Yes, this code path is taken when `streaaming=True` and matches the physical type. For `datetime`s, the physical type and units remain the same, so it simply takes the `Int64` arm.

For `Date` values, the value is upcast to `Int64` in a prior step. So when it hits this section, we need to know to perform the conversion to ms so that, later on when it's re-cast to a `Datetime`, its value is correct. This is also why we need the add additional `Enum` types--to tell the streaming process that we need to go through the date/datetime transformation path when we finalize the mean.